packages/compiler-vapor/__tests__/transforms/vText.spec.ts

```diff
@@ -1,28 +1,123 @@
-import { type RootNode, BindingTypes, DOMErrorCodes } from '@vue/compiler-dom'
-import { type CompilerOptions, compile as _compile } from '../../src'
+import {
+  BindingTypes,
+  DOMErrorCodes,
+  NodeTypes,
+  parse,
+} from '@vue/compiler-dom'
+import {
+  type CompilerOptions,
+  compile as _compile,
+  RootIRNode,
+  transform,
+  generate,
+  IRNodeTypes,
+} from '../../src'
+import { getBaseTransformPreset } from '../../src/compile'
 
-function compile(template: string | RootNode, options: CompilerOptions = {}) {
-  let { code } = _compile(template, {
-    ...options,
-    mode: 'module',
+function compileWithVText(
+  template: string,
+  options: CompilerOptions = {},
+): {
+  ir: RootIRNode
+  code: string
+} {
+  const ast = parse(template, { prefixIdentifiers: true, ...options })
+  const [nodeTransforms, directiveTransforms] = getBaseTransformPreset(true)
+  const ir = transform(ast, {
+    nodeTransforms,
+    directiveTransforms,
     prefixIdentifiers: true,
+    ...options,
   })
-  return code
+  const { code } = generate(ir, { prefixIdentifiers: true, ...options })
+  return { ir, code }
 }
 
 describe('v-text', () => {
-  test('simple expression', () => {
-    const code = compile(`<div v-text="str"></div>`, {
+  test('should convert v-text to textContent', () => {
+    const { code, ir } = compileWithVText(`<div v-text="str"></div>`, {
       bindingMetadata: {
         str: BindingTypes.SETUP_REF,
       },
     })
+
+    expect(ir.vaporHelpers).contains('setText')
+    expect(ir.helpers.size).toBe(0)
+
+    expect(ir.operation).toMatchObject([])
+
+    expect(ir.effect).toMatchObject([
+      {
+        expressions: [
+          {
+            type: NodeTypes.SIMPLE_EXPRESSION,
+            content: 'str',
+            isStatic: false,
+          },
+        ],
+        operations: [
+          {
+            type: IRNodeTypes.SET_TEXT,
+            element: 1,
+            value: {
+              type: NodeTypes.SIMPLE_EXPRESSION,
+              content: 'str',
+              isStatic: false,
+            },
+          },
+        ],
+      },
+    ])
+
     expect(code).matchSnapshot()
   })
 
-  test('no expression', () => {
+  test.fails(
+    'should raise error and ignore children when v-text is present',
```

Comment:
I am trying to fix this test, and then I found that this situation has been handled in the implementation of `v-text`. The reason it didn't take effect is because the transform of prop was placed after the transform of children.

What is the reason for doing this?
And can you provide some guidance on how to solve this problem?

Reply:
Fixed in bc6091b21977b338e9acc21fdb668dcbe6c2f229
Since children have been processed, we should ignore them by setting `context.childrenTemplate`, instead of `node.children`